lib/graphiti/errors.rb

```diff
@@ -816,5 +816,34 @@ def message
 
     class ConflictRequest < InvalidRequest
     end
+
+    class FilterGroupInvalidRequirement < Base
+      def initialize(resource, valid_required_values)
+        @resource = resource
+        @valid_required_values = valid_required_values
+      end
+
+      def message
+        <<-MSG.strip
+          The filter group required: value on resource #{@resource.class} must be one of the following:
+          #{@valid_required_values.join(", ")}
+        MSG
+      end
+    end
+
+    class FilterGroupMissingRequiredFilters < Base
+      def initialize(resource, filter_names, required)
+        @resource = resource
+        @filter_names = filter_names
+        @required_label = required == :all ? 'All' : 'One'
```

Comment:
Do we want to extend this concept to describe and "exclusive or" state? i.e. we want to accept one or the other filter, but not both. I could imagine we could need that as well:

1. `all` -> and
2. `any` -> or
3. `?` -> xor

Does that make sense to you @chiliburger?

Reply:
I think it’s a low hanging fruit for enhancement if and when we need it, but don’t think we need to hold up this feature getting merged either. Leave that up to @chiliburger to decide whether to include or not. As for name, if we do add it, I'd suggest `required: :one`

Reply:
Yes I considered a `:detect` option like in Ruby Enumerable but it would make the implementation a bit more complex to skip filters. Trying to keep it simple at the moment.